core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -67,26 +70,60 @@
 public abstract class AbstractStreamMessageDuplicator<T, U extends StreamMessage<T>>
         implements SafeCloseable {
 
+    private static final long DEFAULT_MAX_SIGNAL_LENGTH = Flags.defaultMaxResponseLength();
```

Comment:
I think you can remove the constructor that uses the default max length and let the subclasses decide the default value.